internal/api/dispatch/impl.go

```diff
@@ -112,14 +112,19 @@ func (this *dispatchManager) ProcessRun(ctx context.Context, account string, ser
 	return entity.ID, correlationID, nil
 }
 
-func (this *dispatchManager) ProcessCancel(ctx context.Context, account string, cancel generic.CancelInput) (runID, correlationID uuid.UUID, err error) {
+func (this *dispatchManager) ProcessCancel(ctx context.Context, orgID string, cancel generic.CancelInput) (runID, correlationID uuid.UUID, err error) {
 	var run db.Run
 
 	if err := this.db.First(&run, cancel.RunId).Error; err != nil {
 		instrumentation.PlaybookRunCancelError(ctx, err)
 		return uuid.UUID{}, run.CorrelationID, &RunNotFoundError{err: err, runID: cancel.RunId}
 	}
 
+	if err := this.db.Where("id = ? AND org_id = ?", cancel.RunId, orgID).First(&run).Error; err != nil {
```

Comment:
This should not be implemented as two queries (more expensive, not transactional, etc.) Instead, either check if `run.OrgID == orgID` or just do `Where("id = ? AND org_id = ?"` right away and return   404 in both cases.